### ISSUE TITLE
Bug in SpeedometerReader

### ISSUE THREAD

**Issue by abol-karimi** [carla-simulator/leaderboard#165]
Is the following intended behavior? The case when `transform` is not assigned (after maximum attempts) doesn't seem to be handled properly.

https://github.com/carla-simulator/leaderboard/blob/111a48f9099c08a2f1068ee8aea2ad56ce52ef9d/leaderboard/envs/sensor_interface.py#L110-L125

**Comment by abol-karimi** [comment]
Sometimes I get the following error:

```
Exception in thread Thread-8:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/home/carla/carla_garage_fork/leaderboard/leaderboard/envs/sensor_interface.py", line 72, in run
    self._callback(GenericMeasurement(self.__call__(), GameTime.get_frame()))
  File "/home/carla/carla_garage_fork/leaderboard/leaderboard/envs/sensor_interface.py", line 125, in __call__
    return {'speed': self._get_forward_speed(transform=transform, velocity=velocity)}
UnboundLocalError: local variable 'transform' referenced before assignment
```

**Comment by abol-karimi** [comment]
The issue seems to be that sometimes the `self._vehicle` is already destroyed so all the attempts to read its velocity and transform fail.

**Comment by glopezdiest** [comment]
Hello @abol-karimi, you are definitely right. We'll do a PR soon fixing that

**Comment by glopezdiest** [comment]
Ups, didn't mean to mark it as completed before. Here is the PR that fixes it  #173
